roxie/ccd/ccdlistener.cpp

```diff
@@ -1768,12 +1768,26 @@ class RoxieProtocolMsgSink : implements IHpccNativeProtocolMsgSink, public CInte
                 Owned<IXmlWriter> logwriter = protocol->writeAppendContent(nullptr);
                 msgctx->writeLogXML(*logwriter);
             }
-            if (statsWuid.length())
+
+            bool summaryStats = msg->getPropBool("@summaryStats", false);
+            if (!statsWuid.isEmpty() || summaryStats)
             {
                 Owned<IXmlWriter> wuwriter = protocol->writeAppendContent(nullptr);
-                wuwriter->outputBeginNested("StatsWorkUnit", true);
-                wuwriter->outputCString(statsWuid.str(), "wuid");
-                wuwriter->outputEndNested("StatsWorkUnit");
+                if (!statsWuid.isEmpty())
+                {
+                    wuwriter->outputBeginNested("StatsWorkUnit", true);
+                    wuwriter->outputCString(statsWuid.str(), "wuid");
+                    wuwriter->outputEndNested("StatsWorkUnit");
+                }
+                if (summaryStats)
+                {
+                    //The query completion time needs discussion and is unavailable for now.
```

Comment:
@ghalliday completion time would be tricky here.  Passing through "start time" is one thing.  but the thing that would need thought is the fact that we are inside of an AsyncFor which could be processing a whole array of requests.  I think ideally this would be the timing for this one array entry,.. but that would be tricky and might be something we wanted to do in a separate Jira?

No matter what, the final completion times would be approximate because we haven't yet finalized or written out the response.

Another field that had to be left off was related to bytes written.  Which we do not know because we are in the middle of finalizing the response before it is written.